pkg/ottl/expression.go

```diff
@@ -738,6 +754,18 @@ func (p *Parser[K]) newGetter(val value) (Getter[K], error) {
 		return &lg, nil
 	}
 
+	if val.Map != nil {
+		mg := mapGetter[K]{mapValues: map[string]Getter[K]{}}
+		for _, kvp := range val.Map.Values {
+			getter, err := p.newGetter(*kvp.Value)
+			if err != nil {
+				return nil, err
+			}
+			mg.mapValues[*kvp.Key] = getter
+		}
+		return &mg, nil
```

Comment:
```suggestion
		return StandardPMapGetter[K]{Getter: &mg}, nil
```
This could be another way to return a `pcommon.Map`.

Reply:
unfortunately the `StandardPMapGetter` can not be used here, as it's `Get` function returns a `pcommon.Map`, whereas the `newGetter` method is expected to return an implementation of the `Getter` interface which requires its `Get` method to return `any`. However, I found a way to modify the `mapGetter` to return a `pcommon.Map` now.